core/src/main/java/org/apache/accumulo/core/constraints/Constraint.java

```diff
@@ -92,4 +84,16 @@
    * @return list of violation codes, or null if none
    */
   List<Short> check(Environment env, Mutation mutation);
+
+  /**
+   * For backwards compatibility. New API equivalent of {@link #check(Environment, Mutation)} but
+   * renamed to prevent ambiguous method call errors.
+   *
+   * @since 2.1.0
+   */
+  @Override
+  default List<Short> checkMutation(
+      org.apache.accumulo.core.data.constraints.Constraint.Environment env, Mutation mutation) {
+    throw new UnsupportedOperationException();
+  }
```

Comment:
I am not seeing where the tserver would ever call the older check() method for code implementing the older interface.  This method could do that serving as a bridge to code implementing the older interface.

```suggestion
  default List<Short> checkMutation(
      org.apache.accumulo.core.data.constraints.Constraint.Environment env, Mutation mutation) {
    return check(SomeClassNotInPublicAPI.convert(env), mutation);
  }
```

Reply:
If the class that implements Environment implements the old and new interface, then it could just be cast in this method.  Like

```java
default List<Short> checkMutation(...) {
    return check((Environment)env, mutation);
}
```

Reply:
I think this is where the `check()` method gets called today. I had changed this to call checkMutation():
https://github.com/apache/accumulo/blob/9b504121dd69bf033dced56adb55d727b0eafdc4/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java#L1062